commands/pre-push

```diff
@@ -11,11 +11,11 @@ REFS=$(cat)
 parent_pid=$PPID
 push_command=''
 
-#until [[ "$push_command" == git* ]]; do
-#  push_command=$(ps -ocommand= -p $parent_pid)
-#  parent_pid=$(ps -o ppid= $parent_pid)
-#  # echo "command = $push_command"
-#done
+until [[ "$push_command" == git* ]]; do
```

Comment:
I had a look at this again and this section you commented out is all just debugging variables really.
It was a script I adapted from some [online source](https://stackoverflow.com/a/63322932/11587288)

> The script basically will do 2 things:
This script will block anybody who tries to push to a certain branch (in my case I don't want anybody -including myself- to push directly to master and develop branch). They need to work in their own branch and then create a pull request.
This script will block anybody who tries to push to a branch that is different from their current active branch. For example you are in branch fix/someissue but then you mistakenly type git push origin master.


"Why use this over protected branches"
Good question. I can't think of anything at present

Although I have dim memories of discussing this with @traviscrist. 
It may have been something to do with the way our PR merge permissions are configured and some Github inflexibility (which may or may not exist any more)

Reply:
I got the script working on my local machine and added how to the issue you linked previously. I still think we might consider protected branches, but I'm okay with leaving this how it is currently.